components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -996,19 +1035,19 @@ void FeedV2Builder::UpdateData(UpdateSettings settings,
   }
 
   if (settings.signals) {
-    signals_.clear();
```

Comment:
It looks like all the `raw_feed_items_/suggested_publisher_ids_/topics_/etc` could be grouped into a one struct and be passed/cleared/updated synchronously.